nemo_curator/synthetic/nemotron.py

```diff
@@ -0,0 +1,1472 @@
+                f"Error: Length of macro topics {len(macro_topics)} does not match desired n_macro_topics {n_macro_topics}: {macro_topics}"
+            )
+        macro_topics.extend(additional_macro_topics)
+
+        # Generate the subtopics
+        raw_topics = [
+            self.generate_python_subtopics(
+                macro_topic=macro_topic,
+                n_subtopics=n_subtopics,
+                model=model,
+                model_kwargs=base_model_kwargs,
+                prompt_template=subtopic_prompt_template,
+            )[0]
+            for macro_topic in macro_topics
+        ]
+        topic_list = []
+        for topic in raw_topics:
+            try:
+                parsed_topics = self.convert_response_to_yaml_list(
+                    topic,
+                    model=model,
+                    prompt_template=yaml_conversion_prompt_template,
+                    model_kwargs=conversion_model_kwargs,
+                )
+                if len(parsed_topics) != n_subtopics:
+                    raise YamlConversionError(
+                        f"Error: Length of subtopics {len(parsed_topics)} does not match desired n_subtopics {n_subtopics}: {parsed_topics}"
+                    )
+                topic_list.extend(parsed_topics)
+            except YamlConversionError as e:
+                if ignore_conversion_failure:
+                    continue
+                else:
+                    raise e
+        topic_list.extend(additional_subtopics)
+
+        # Mix the macro topics with the subtopics
+        if combine_topics:
+            topic_list.extend(macro_topics)
+
+        # Generate the openlines
+        raw_lines = [
+            self.generate_python_problem(
+                topic=t,
+                n_openlines=n_openlines,
+                model=model,
+                model_kwargs=base_model_kwargs,
+                prompt_template=python_problem_prompt_template,
+            )[0]
+            for t in topic_list
+        ]
+        openlines = []
+        for line in raw_lines:
+            try:
+                parsed_line = self.convert_response_to_yaml_list(
+                    line,
+                    model=model,
+                    prompt_template=yaml_conversion_prompt_template,
+                    model_kwargs=conversion_model_kwargs,
+                )
+                if len(parsed_line) != n_openlines:
+                    raise YamlConversionError(
+                        f"Error: Length of openlines {len(parsed_line)} does not match desired n_openlines {n_openlines}: {parsed_line}"
+                    )
+                openlines.extend(parsed_line)
+            except YamlConversionError as e:
+                if ignore_conversion_failure:
+                    continue
+                else:
+                    raise e
+
+        return openlines
+
+
+class NemotronFormatter(ConversationFormatter):
+
+    BASE_PROMPT = "<extra_id_0>System\n\n<extra_id_1>User\n"
```

Comment:
Should this be in prompts.py ?

Reply:
No, I don't think so. This isn't really useful outside of a conversational formatter imo. I changed the name to be a bit less confusing.

Reply:
Ahh, cool, thanks for the variable rename. Threw me off.